libp2p/connmanager.nim

```diff
@@ -307,6 +307,9 @@ proc peerCleanup(c: ConnManager, conn: Connection) {.async.} =
     await c.triggerConnEvent(
       peerId, ConnEvent(kind: ConnEventKind.Disconnected))
     await c.triggerPeerEvents(peerId, PeerEvent(kind: PeerEventKind.Left))
+
+    if not(c.peerStore.isNil):
+      c.peerStore.cleanup(peerId)
```

Comment:
It makes sense to clean up peers that were disconnected. If Waku were to use the native `switch` peer store, we'd need to change some peer persistence logic though - we do currently track disconnected peers in our peer store (a `DisconnectBook`), mostly because it's a convenient way to determine a back off period before attempting to reconnect to such peers after a restart. I think we need to do something else here in any case, though, and can't think of any other good reasons to keep peers in the store after disconnection.

Reply:
Yes, currently this will keep a certain amount of disconnected peers (peerStore.capacity which the app can choose), and then remove them from every book.

I was wondering if you needed more sophisticated logic (eg, have a per-book settings on how much dead peer to save)
Happy to brainstorm a bit on your needs :)